scripts/verify-replace.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+TMP_DIR=/tmp/
+BASE_REPO_PATH=$(mktemp -d ${TMP_DIR}replace-verify.XXX)
+GH_BASE_URL_KS=https://github.com/kubesaw/
+GH_BASE_URL_CRT=https://github.com/codeready-toolchain/
+declare -a repos=("${GH_BASE_URL_KS}ksctl" "${GH_BASE_URL_CRT}host-operator" "${GH_BASE_URL_CRT}member-operator" "${GH_BASE_URL_CRT}registration-service" "${GH_BASE_URL_CRT}toolchain-e2e")
+C_PATH=${PWD}
+ERRORLIST=()
+
+echo Initiating verify-replace on dependent repos
+for repo in "${repos[@]}"
+do
+    echo =========================================================================================
+    echo  
+    echo                        "$(basename $repo)"
+    echo                                                                     
+    echo =========================================================================================                                            
+    REPO_PATH=$BASE_REPO_PATH/$(basename $repo)
+    echo "Cloning repo in /tmp"
+    git clone --depth=1 $repo $REPO_PATH
+    echo "Repo cloned successfully"
+    cd $REPO_PATH || exit
+    echo "Initiating 'go mod replace' of current toolchain common version in dependent repos"
+    go mod edit -replace github.com/codeready-toolchain/toolchain-common=$C_PATH
+    make verify-dependencies || ERRORLIST+="($(basename $repo))"
+    echo                                                          
+    echo =========================================================================================
+    echo                                                           
+done
+if [ ${#ERRORLIST[@]} -ne 0 ]; then
+    echo "Below are the repos with error: "
+    for e in ${ERRORLIST[*]}
+    do
+        echo "$e"
+    done
```

Comment:
we should exit with a non-zero value if something failed, right?

Reply:
not sure if we require the non zero exit here ?, 
here we are just displaying the error  repo names, and error list is already being checked if its empty or not , so my understanding was we dont require exit here.? 
Probably i am missing something here 🤔

Reply:
if one of the verification failed, then we want the whole makefile target to return non-zero value (in other words let it fail as well)

Reply:
updated it